src/operator/image/image_random.cu

```diff
@@ -99,18 +111,242 @@ void ToTensorImplCUDA(mshadow::Stream<gpu> *s,
+                            const Tensor<xpu, 3, DType> in_grad,
+                            const int req,
+                            const int N,
+                            const int H,
+                            const int W,
+                            const int C,
+                            const float std_d0,
+                            const float std_d1,
+                            const float std_d2) {
+    // We process one image per thread block.
+    // In 3D case, we have only 1 block i.e., blockIdx.x
+    // We do not use it.
+
+    float std = std_d0;
+    for (int c = 0; c < C; ++c) {
+        switch (c) {
+            case 0 : std = std_d0;
+                     break;
+            case 1 : std = std_d1;
+                     break;
+            case 2 : std = std_d2;
+                     break;
+        }
+        for (int h = threadIdx.y; h < H; h += blockDim.y) {
+            for (int w = threadIdx.x; w < W; w += blockDim.x) {
+                KERNEL_ASSIGN(in_grad[c][h][w], req,
+                              out_grad[c][h][w] * (1 / std));
+            }
+        }
+    }
+}
+
+// Normalize Backward Kernel for 3D input
+template<typename xpu, typename DType>
+__global__ void
+__launch_bounds__(cuda::kMaxThreadsPerBlock, 1)
+NormalizeBackwardCudaKernel(const Tensor<xpu, 4, DType> out_grad,
+                            const Tensor<xpu, 4, DType> in_grad,
+                            const int req,
+                            const int N,
+                            const int H,
+                            const int W,
+                            const int C,
+                            const float std_d0,
+                            const float std_d1,
+                            const float std_d2) {
+    // We process one image per thread block.
+    const int n = blockIdx.x;
+
+    float std = std_d0;
+    for (int c = 0; c < C; ++c) {
+        switch (c) {
+            case 0 : std = std_d0;
+                     break;
+            case 1 : std = std_d1;
+                     break;
+            case 2 : std = std_d2;
+                     break;
+        }
+        for (int h = threadIdx.y; h < H; h += blockDim.y) {
+            for (int w = threadIdx.x; w < W; w += blockDim.x) {
+                KERNEL_ASSIGN(in_grad[n][c][h][w], req,
+                              out_grad[n][c][h][w] * (1 / std));
+            }
+        }
+    }
+}
+
+template<typename DType, typename T>
+void NormalizeBackwardImplCUDA(mshadow::Stream<gpu> *s,
+                               const T out_grad,
+                               const T in_grad,
+                               const int req,
+                               const float std_d0,
+                               const float std_d1,
+                               const float std_d2) {
+    int blocks, H, W, C, N;
+    cudaStream_t stream = mshadow::Stream<gpu>::GetStream(s);
+    if (std::is_same<T, Tensor<gpu, 3, DType>>::value) {
```

Comment:
construct a 4D tensor (1, C, H, W) by reusing the `data_` pointer of 3D tensor (C, H, W), delete all 3D cuda kernels and specifications

Reply:
Done.